x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportPutDataFrameTransformAction.java

```diff
@@ -213,9 +212,15 @@ private void putDataFrame(DataFrameTransformConfig config, ActionListener<Acknow
         // <2> Put our transform
         ActionListener<Boolean> pivotValidationListener = ActionListener.wrap(
             validationResult -> dataFrameTransformsConfigManager.putTransformConfiguration(config, putTransformConfigurationListener),
-            validationException -> listener.onFailure(
-                new RuntimeException(DataFrameMessages.REST_PUT_DATA_FRAME_FAILED_TO_VALIDATE_DATA_FRAME_CONFIGURATION,
-                    validationException))
+            validationException -> {
+                if (validationException instanceof Pivot.TestQueryException && request.isDeferValidation()) {
```

Comment:
So are we going to defer selected validations on a case-by-case basis? Or do you follow a concrete rule that says which validations are ok to be deferred?

Reply:
This is coming from `Pivot.validate()` just off the screen isn't it? Why not add a `run_test_query` parameter to that method or break up `Pivot.validate()` to do the supported aggs check and test query in different methods.

Reply:
+1 for splitting `Pivot.validate()` into syntax (or config) validation and runtime validation